open-reac/src/test/java/com/powsybl/openreac/parameters/input/BranchImpedanceValidationTest.java

```diff
@@ -0,0 +1,440 @@
+/**
+ * Copyright (c) 2025, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+package com.powsybl.openreac.parameters.input;
+
+import com.powsybl.commons.report.ReportNode;
+import com.powsybl.iidm.network.*;
+import com.powsybl.openreac.exceptions.InvalidParametersException;
+import org.junit.jupiter.api.Test;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+/**
+ * @author Oscar Lamolet {@literal <lamoletoscar at proton.me>}
+ */
+class BranchImpedanceValidationTest {
+
+    /**
+     * Create a ReportNode for testing
+     */
+    private ReportNode createReportNode() {
+        return ReportNode.newRootReportNode()
+                .withMessageTemplate("branchImpedanceValidationTest", "Branch Impedance Validation Test")
+                .build();
+    }
+
+    /**
+     * Test that French branches with r > 10*|x| throw an error
+     */
+    @Test
+    void testFrenchBranchWithVeryHighImpedanceRatioThrowsError() {
+        Network network = createNetworkWithFrenchLine(12.0, 1.0); // ratio = 12 > 10
+
+        OpenReacParameters params = new OpenReacParameters();
+        ReportNode reportNode = createReportNode();
+
+        InvalidParametersException exception = assertThrows(InvalidParametersException.class,
+                () -> params.checkIntegrity(network, reportNode));
+
+        assertTrue(exception.getMessage().contains("French branches have r > 10 * |x|"));
+        assertTrue(exception.getMessage().contains("Line 'LINE_FR'"));
+    }
+
+    /**
+     * Test that French branches with 1 < r/|x| <= 10 generate a warning but don't throw
+     */
+    @Test
+    void testFrenchBranchWithModerateImpedanceRatioGeneratesWarning() {
+        Network network = createNetworkWithFrenchLine(5.0, 1.0); // ratio = 5, acceptable
+
+        OpenReacParameters params = new OpenReacParameters();
+        ReportNode reportNode = createReportNode();
+
+        // Should not throw
+        assertDoesNotThrow(() -> params.checkIntegrity(network, reportNode));
+
+        // Should have warning in ReportNode
+        boolean hasWarning = hasReportWithKey(reportNode, "optimizer.openreac.nbFrenchBranchesWithAcceptableHighImpedanceRatio");
```

Comment:
`VoltageLevelLimitsOverrideInputTest` already has a similar method:
`private static boolean checkReportWithKey(String key, ReportNode reportNode)`

To avoid duplication, can you move it in a new static class (in the test directory) and use it instead?